static/js/popovers.js

```diff
@@ -480,31 +480,19 @@ exports.register_click_handlers = function () {
         e.preventDefault();
     });
     $('body').on('click', '.popover_toggle_collapse', function (e) {
-        var home_row;
         var msgid = $(e.currentTarget).data('msgid');
         var row = current_msg_list.get_row(msgid);
         var message = current_msg_list.get(rows.id(row));
 
-        // If we are narrowed we also need to collapse this message in the home
-        // view.
-        if (current_msg_list === narrowed_msg_list) {
-            home_row = home_msg_list.get_row(msgid);
-        }
-
-        var toggle_row = function toggle_row(row) {
-            if (!row) { return; }
-
-            if (message.collapsed) {
-                condense.uncollapse(row);
-            } else {
-                condense.collapse(row);
-            }
-        };
-
         popovers.hide_actions_popover();
 
-        toggle_row(row);
-        toggle_row(home_row);
+        if (!row) { return; }
```

Comment:
I think this line is not correct, since the `return` will escape from this function, whereas previously it would have just escaped `toggle_row`.  The end result is the `e.stopPropagation` pair of lines won't be run in your new version.

Probably should just instead move the `if (message.collapsed)` block below inside an `if (row) {`...